pkg/controllers/provisioning/provisioner.go

```diff
@@ -206,7 +206,7 @@ func (p *Provisioner) schedule(ctx context.Context, pods []*v1.Pod) ([]*schedule
 		return nil, fmt.Errorf("getting daemon overhead, %w", err)
 	}
 
-	return scheduler.NewScheduler(nodeTemplates, p.cluster, topology, instanceTypes, daemonOverhead, p.recorder).Solve(ctx, pods)
+	return scheduler.NewScheduler(nodeTemplates, provisionerList.Items, p.cluster, topology, instanceTypes, daemonOverhead, p.recorder).Solve(ctx, pods)
```

Comment:
Thoughts on generating the node templates inside the scheduler, now that we're passing the provisioner in?

Reply:
NodeTemplates are used by topology and getDaemonOverhead which are both in the provisioner.  The NodeTemplate being a subset of a provisioner with the added functionality of a tightened set of requirements causes some complications. If you need the requirements, you need the NodeTemplate.  If you need anything outside the subset that's in NodeTemplate, you need both the provisioner and the NodeTemplate.